### DISCUSSION TITLE
OmvDataSource handling of HTTP 204

### DISCUSSION THREAD

**Discussion by brent-williams** [heremaps/harp.gl#2210] [Q&A] [answered]
Hi harp.gl folks I have a question about zooming.  I have a data source that contains tiles up to level 12 and returns 204 for level 13+.  I'd expect harp.gl to render the level 12 tiles when zooming at level 13+.  However it seems harp.gl simply does not render the data source at level 13.  

Is there a config somewhere so that harp.gl renders lower-zoom tiles when higher-zoom tiles are not available?

Thanks for any tips.  Regards Brent

**Comment by nzjony** [comment, marked as answer]
hi @brent-williams , thanks for the great question.

You can specify a `maxDataLevel` on the DataSource. I.e. if you configured it with level 12, then levels greater than that (up to some limit, see https://www.harp.gl/docs/master/doc/interfaces/harp_mapview.mapviewoptions-1.html#quadtreesearchdistancedown and https://www.harp.gl/docs/master/doc/modules/_docs_fallback.html) will just show the levels 12 tile.

As far as I can tell, there is no such logic to handle empty tiles and fallback to high level tiles automatically (i.e. without using `maxDataLevel`).

Here we create the tile from the datasource I.e. see: https://github.com/heremaps/harp.gl/blob/df1b40df93c33bb851f93b6069629622f4566999/%40here/harp-mapview/lib/VisibleTileSet.ts#L1283, then on line 1286 we add the tile to the queue to be processed and then we add it to the cache.

Part of the problem is this here: https://github.com/heremaps/harp.gl/blob/df1b40df93c33bb851f93b6069629622f4566999/%40here/harp-mapview/lib/Tile.ts#L724 Tile's that have no geometry are forced to have geometry. I'm not exactly sure why, it should be investigated.

Hopefully the `maxDataLevel` works for now?

I will create an internal ticket to track this.

**Reply by brent-williams** [reply]
Excellent, maxDataLevel is exactly what I was looking for, thanks for the quick response!

**Reply by nzjony** [reply]
You're welcome!
